TACC-Cloud/hazmapper
# task/WG-250: Updating project creation structure

## Overview: ##
Updating project creation structure to match changes from WG-212

## PR Status: ##


## Related Jira tickets: ##

* [WG-250](https://tacc-main.atlassian.net/browse/WG-250)

## Summary of Changes: ##

- Updated projects.ts under the types directory to reflect new payload and response structures for project creation. This included getting rid of title and streetview_instances props in the response. It also included renaming observable to watch_users in the request/payload. 
- Updated the projectData structure (data being submitted in the creation) in CreateMapModal comp to match these changes. 
- Updated the unit test for CreateMapModal to reflect the new structure. We no longer have a project object with nested props. Now it's all one ProjectRequest object. 

## Testing Steps: ##
1. This needs to be run with the local geoapi backend (master branch) to work. 
2. Go to secrets.local and change backend to Local
3. npm ci
4. npm run dev
5. Click Create Map button
6. Create a new map project
7. Make sure there's no errors with the POST request and the map loads (default tile layers created, file created, etc)

## UI Photos:

## Notes: ##
